### PR TITLE
:seedling: Replace deprecated functions from k8s.io/utils/pointer

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon <kmuldoon@vmware.com>

Replace deprecated functions in https://pkg.go.dev/k8s.io/utils/pointer with the replacement functions.

The Kubernetes util package has deprecated all of the functions named *Ptr e.g. pointer.StringPtr(), pointer.IntPtr(), pointer.BoolPtr() and replaced them with equivalent functions named by their simple type e.g. pointer.String(), pointer.Int(), pointer.Bool().

/kind cleanup

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm

we did that in k/k for kubeadm

File: api/v1alpha3/cluster_types.go [hunk 1]
```diff
@@ -255,7 +255,7 @@ func (in FailureDomains) FilterControlPlane() FailureDomains {
 func (in FailureDomains) GetIDs() []*string {
 	ids := make([]*string, 0, len(in))
 	for id := range in {
-		ids = append(ids, pointer.StringPtr(id))
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply api/v1alpha3/cluster_types.go:258]: Totally fine. Just an implementation detail of the func

**Comment by sbueringer** [comment]
Thx!

/approve

Slightly wondering why we didn't have linter findings for using deprecated APIs

**Comment by sbueringer** [comment]
@killianmuldoon I think staticcheck was verifying that in another project I was using a while back.

**Pull request merged**